src/executables/elm_ats_api/elm_ats_driver.hh

```diff
@@ -0,0 +1,113 @@
+/*
+ELM-ATS driver:
+Provides ATS functionality to an external caller like ELM
+contains:
+setup
+initialize
+advance
+set_sources
+get_waterstate
+get_mesh_info
+*/
+
+#pragma once
+
+#include "Teuchos_RCP.hpp"
+#include "Mesh.hh"
+#include "MeshPartition.hh"
+#include "Key.hh"
+#include "coordinator.hh"
+
+namespace ATS {
+
+class ELM_ATSDriver : public Coordinator {
+
+public:
+  
+  ELM_ATSDriver();
+  ~ELM_ATSDriver() = default;
+
+  // setup and advance are overloads, but not overrides, of the
+  // setup and advance functions in the base Coordinator class
+  // they have the same name, but the signatures are different,
+  // so they aren't virtual and they don't override Coordinator
+  void setup(MPI_Fint *f_comm, const char *input_filename);
```

Comment:
Can we add a "verbosity" flag to this as well?  It would be nice to set the global verbosity (typically done through the command line) based on some ELM flag.  

I guess alternatively we could add a global verbosity flag to the ATS input file, which we probably either already have or should have.